sw/template/src/main.c

```diff
@@ -2,13 +2,28 @@
 
 int main(){
 
-    uint32_t * gpio_addr = (uint32_t *) 0x100000;
+    /* Insert your code here */
+    
+    // uint32_t * gpio_addr = (uint32_t *) 0x100000;
+
+    // while(1){
+	// for(int i = 0; i < 100000; i++);		
+    // 	*gpio_addr = 0xffffffff;
+    // 	for(int i = 0; i < 100000; i++);		
+    // 	*gpio_addr = 0x00000000;
+    // }
+
+
+    uint32_t * uart = (uint32_t *) 0x00001000;
+    char * str = "Hello world!!!";
+    uint8_t i = 0;
 
     while(1){
-	    for(int i = 0; i < 100000; i++);
-    	*gpio_addr = 0x00000001;
-    	for(int i = 0; i < 100000; i++);
-    	*gpio_addr = 0x00000000;
+        if (i<14) {
+            while( (*(uart+2)) != 0x04000000);  /* Wrong endianess - it has to be 0x00000004 */
```

Comment:
Export these literals in constants or macros, with this note about endianess. 

This way the functionality and the endianess workaroud are logically split and easier to fix once solved.

Reply:
Done, but int the same source (?)